src/python/pants/core/environments/rules.py

```diff
@@ -764,20 +766,39 @@ def _add_option_field_for(
                 "value type."
             )
 
-    # The below class will never be used for static type checking outside of this function.
-    # so it's reasonably safe to use `ignore[name-defined]`. Ensure that all this remains valid
-    # if `_SIMPLE_OPTIONS` or `_LIST_OPTIONS` are ever modified.
-    class OptionField(field_type, _EnvironmentSensitiveOptionFieldMixin):  # type: ignore[valid-type, misc]
-        alias = f"{scope}_{snake_name}".replace("-", "_")
-        required = False
-        value: Any
-        help = (
+    class_attrs = {
+        "alias": f"{scope}_{snake_name}".replace("-", "_"),
+        "required": False,
+        "help": (
             f"Overrides the default value from the option `[{scope}].{snake_name}` when this "
             "environment target is active."
-        )
-        subsystem = env_aware_t
-        option_name = option.args[0]
-
+        ),
+        "subsystem": env_aware_t,
+        "option_name": option.args[0],
+    }
+
+    # For sequence fields, use special handling to distinguish "not specified" from "explicitly empty"
+    if issubclass(field_type, SequenceField):
+        class_attrs["none_is_valid_value"] = True
+
+        def compute_value(cls, raw_value: Any | None, address: Address) -> Any | None:
+            # When field is not specified in BUILD file, return None to indicate inheritance
+            if raw_value is NO_VALUE:
+                return None
+            # When explicitly set to empty list, return empty tuple (don't inherit)
+            elif raw_value == []:
+                return ()
```

Comment:
Can you confirm why this needs special handling?

My initial reaction would be that there's already no inheritance for `[]`... or at least, if there is, I don't understand which code causes it to happen (e.g. `resolve_environment_sensitive_option` and its callers like `EnvironmentAware.__getattribute__` seem to be quite careful about checking for `None` specifically, rather than just relying on truthiness).

(Assuming it's necessary, do we also need to handle `raw_value == ()` in case a user has specified `()` in their `BUILD` file?)